@trpc/next-layout/client.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from "@tanstack/react-query";
+import { createTRPCClient, httpBatchLink } from "@trpc/client";
+import { AnyRouter, ProcedureType } from "@trpc/server";
+import { createRecursiveProxy } from "@trpc/server/shared";
+import { createContext } from "react";
+import { DecoratedProcedureRecord } from "./createTRPCNextLayout";
+import superjson from "superjson";
+export type TRPCSchemaContext = Record<string, ProcedureType>;
+export const TRPCSchemaContext = createContext<TRPCSchemaContext>(null as any);
+export const TRPCSchemaContextProvider = TRPCSchemaContext.Provider;
+
+export async function createTRPCNextClientLayout<
+  TRouter extends AnyRouter
+>(): Promise<DecoratedProcedureRecord<TRouter["_def"]["record"]>> {
+  const client = createTRPCClient({
+    links: [httpBatchLink({ url: "/api/trpc" })],
```

Comment:
On the server I think we'd need some sort of server-link?

Reply:
This wouldn't run on the server. `@trpc/next-layout/server.ts` is the one that runs on the server